### PR TITLE
Build out account sidebar

### PR DESCRIPTION
**Pull request by JoshAntBrown**
I've put this together for the account sidebar. Should be a near 1 to 1 match with Figma since I checked all the dimensions to the tailwind ones or used a custom number where required.

I'm not particularly happy about the way I've queried the accounts within the partial right now, it might be nicer if we could just do `type.all` and have the model scoped by the Current.family automatically but that's a wider problem than the scope of this issue.

/claim #386

### CODE REVIEW

File: app/views/accounts/_account_list.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<%# locals: (type:) -%>
+
+<% accounts = Current.family.accounts.where(accountable_type: type.name) %>
+
+<div class="text-sm mb-1" data-controller="reveal">
+  <button class="flex gap-4 px-2 py-3 items-center w-full rounded-[10px] font-medium hover:bg-[#f2f2f2]" data-action="reveal#toggle">
```
Comment (robzolkos) [inline app/views/accounts/_account_list.html.erb:6]: @JoshAntBrown any thoughts on using summary/detail element for this?

Comment (JoshAntBrown) [reply app/views/accounts/_account_list.html.erb:6]: Good call, will refactor, wanted to get a first version out

Comment (JoshAntBrown) [reply app/views/accounts/_account_list.html.erb]: Much better with the detail element, no need for the stimulus controller, and toggling between the icons feels more robust being based on the state of the dialog. Thanks!

**Comment by Shpigford** [comment]
@JoshAntBrown Would you mind adding a GIF or short video showing the functionality? Just a super quick little thing to show that it's functioning. 🙂 Much easier to give feedback on that side of things without pulling the code down.

File: app/views/accounts/_account_list.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<%# locals: (type:) -%>
+
+<% accounts = Current.family.accounts.where(accountable_type: type.name) %>
+
+<details class="text-sm mb-1 group">
+  <summary class="flex gap-4 px-2 py-3 items-center w-full rounded-[10px] font-medium hover:bg-[#f2f2f2]" data-action="reveal#toggle">
```
Comment (robzolkos) [inline app/views/accounts/_account_list.html.erb:6]: Is this data-action still needed?

**Comment by Shpigford** [comment]
@JoshAntBrown Amazing! Looks good to me. Merging now.

**Pull request merged**